### PR TITLE
feat(checkout-sessions): add cart items and metadata to checkout session [TA-4949]

### PR DESCRIPTION
**Pull request by douglaseggleton**
# Description

Adds `metadata` and `cart_items` to checkout sessions.

https://gr4vy.atlassian.net/browse/TA-4879

## Checklist

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own changes
- [x] I have run `yarn test` to make sure my changes pass all linters
- [x] I have pulled the latest changes from the upstream `main` branch

## Contribution guidelines

For contribution guidelines, styleguide, and other helpful information please
see the `CONTRIBUTING.md` file in the root of this project.

### CODE REVIEW

**Comment by cbetta** [comment]
@douglaseggleton perfect, it's exactly what I had in mind.

It might be good to talk to @bloowitt as well. He's working on a change to the Cart Items at the moment in CoreAPI.

File: reference/resources/checkout_session.yaml [hunk 1]
```diff
@@ -27,3 +27,25 @@ properties:
...
       By default this will be set to 1 hour from the date of creation.
     example: "2022-01-01T00:00:00+00:00"
+
+  cart_items:
+    type: array
+    description: |-
+      An array of cart items that represents the line items of a transaction.
+    items:
+      $ref: "../openapi.yaml#/components/schemas/CartItem"
+    maxItems: 249
```
Comment (cbetta) [inline reference/resources/checkout_session.yaml:37]: I think there's no limit anymore now. Right @bloowitt ?

```suggestion
```

File: reference/request-bodies/checkout_session_update_request.yaml [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+  - Checkout
+
+properties:
+  cart_items:
+    type: array
+    description: |-
+      An array of cart items that represents the line items of a transaction.
+    items:
+      $ref: "../openapi.yaml#/components/schemas/CartItem"
+    maxItems: 249
```
Comment (cbetta) [inline reference/request-bodies/checkout_session_update_request.yaml:16]:
```suggestion
```

**Review by cbetta — CHANGES_REQUESTED** [review]
I think we can drop the 249 limit now on cart items since this week.

**Comment by cbetta** [comment]
@douglaseggleton alex is out this week. I checked CoreAPI and we indeed dropped the 249 limit

File: reference/request-bodies/checkout_session_create_request.yaml [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+  - Checkout
+
+properties:
+  cart_items:
+    type: array
+    description: |-
+      An array of cart items that represents the line items of a transaction.
+    items:
+      $ref: "../openapi.yaml#/components/schemas/CartItem"
+    maxItems: 249
```
Comment (cbetta) [inline reference/request-bodies/checkout_session_create_request.yaml:16]: This can go

Comment (douglaseggleton) [inline reference/request-bodies/checkout_session_create_request.yaml:16]:
```suggestion
```

**Review by cbetta — APPROVED** [review]

**Pull request merged**